### DISCUSSION TITLE
Key Auth apiKey on the Declarative Config

### DISCUSSION THREAD

**Discussion by rendyfebry** [Kong/kong#7326] [Ideas and feature requests]
### Summary

When we use Kong in DBLess mode, the `apiKey` for the key-auth plugin should be specified on the declarative config file, that's not ideal from a security perspective.

The other option is to just let Kong generate the key, this also not ideal since the key always changes after Kong restarted.

A Proposed solution maybe makes the declarative config file support env variable, so the key can be injected on runtime. Or somehow just make the key for the same consumer didn't change on every restart.

---

I also tried to find the exact code when the app actually saves the key from the declarative config to the DB, but I can't found them. That weird since I also unable to found `KONG_DECLARATIVE_CONFIG` anywhere on `Kong/kong` repo. Is the code on the private repo for enterprise only?

**Comment by rendyfebry** [comment]
Just want to update, I'm able to get around this by using a custom docker entry point script.

**Comment by ff6347** [comment]
> Just want to update, I'm able to get around this by using a custom docker entry point script.

@rendyfebry can you share an example for the custom entrypoint script?

**Comment by dqii** [comment]
+1^
